ironfish/src/wallet/account/encoder/generic.ts

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { AccountImport } from '../../walletdb/accountValue'
+import { Bech32Encoder } from './bech32'
+import { Bech32JsonEncoder } from './bech32json'
+import {
+  AccountDecodingOptions,
+  AccountEncoder,
+  AccountEncodingOptions,
+  Formats,
+} from './encoder'
+import { JsonEncoder } from './json'
+import { MnemonicEncoder } from './mnemonic'
+import { SpendingKeyEncoder } from './spendingKey'
+
+export class GenericEncoder implements AccountEncoder {
+  versions = [
+    JsonEncoder,
+    Bech32JsonEncoder,
+    Bech32Encoder,
+    MnemonicEncoder,
+    SpendingKeyEncoder,
+  ]
+
+  encode(value: AccountImport, options: AccountEncodingOptions = {}): string {
+    switch (options.format) {
+      case Formats.JSON:
+        return new JsonEncoder().encode(value)
+      case Formats.Bech32:
+        return new Bech32Encoder().encode(value)
+      case Formats.SpendingKey:
+        return new SpendingKeyEncoder().encode(value)
+      case Formats.Mnemonic:
+        return new MnemonicEncoder().encode(value, options)
+      default:
+        return new Bech32Encoder().encode(value)
+    }
+  }
+
+  decode(value: string, options: AccountDecodingOptions = {}): AccountImport {
+    let decoded = null
+    const decoderErrors: [string, Error][] = []
+    for (const encoder of this.versions) {
+      try {
+        decoded = new encoder().decode(value, options)
+      } catch (e) {
+        decoderErrors.push([encoder.name, e as Error])
+        continue
+      }
+      return decoded
+    }
+    const errors = decoderErrors.map(([str, err]) => `${str}: ${err.message}`)
```

Comment:
Had to add this, otherwise the error could be very cryptic. IE if the account is input is `mnemonic` and they didn't provide `name` value to RPC, this would fail cryptically even though the mnemonic value might be correct.